link/link_test.go

```diff
@@ -130,6 +140,27 @@ func mustCgroupEgressProgram(t *testing.T) *ebpf.Program {
 	return prog
 }
 
+func mustCgroupEgressProgramWithMap(t *testing.T, arr *ebpf.Map) *ebpf.Program {
+	t.Helper()
+
+	prog, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Type:       ebpf.CGroupSKB,
+		AttachType: ebpf.AttachCGroupInetEgress,
+		License:    "MIT",
+		Instructions: asm.Instructions{
+			asm.LoadMapPtr(asm.R1, arr.FD()),
+			asm.Mov.Imm(asm.R2, 0),
+			asm.FnGetLocalStorage.Call(),
```

Comment:
I'm assuming that this will only accept a local storage map? If that's true it would make sense to  move creating of the array here.

Reply:
Absolutely. I'll clean this up.